### PR TITLE
Add piece model and inherit individual pieces

### PR DESCRIPTION
**Pull request by ZachBerman**
STI dictates adding a type column to our pieces table

-All chess pieces (pawn, rook, knight, bishop, queen, king) inherit from piece model
-Piece model validates the above strings are allowed only

### CODE REVIEW

File: app/models/piece.rb [hunk 1]
```diff
@@ -0,0 +1,6 @@
+class Piece < ActiveRecord::Base
+  belongs_to :game
+  belongs_to :player
+
+  validates :type, inclusion: { in: %w(pawn rook bishop king queen) }
```
Comment (lasley) [inline app/models/piece.rb]: Missing knight

File: spec/models/knight_spec.rb [hunk 2]
```diff
@@ -0,0 +1,5 @@
+require 'rails_helper'
+
+RSpec.describe Knight, type: :model do
```
Comment (lasley) [inline spec/models/knight_spec.rb]: You should be testing the creations of all your children. This test would have yielded a failure due to lack of knight in type validation

File: app/models/piece.rb [hunk 3]
```diff
@@ -0,0 +1,6 @@
+class Piece < ActiveRecord::Base
+  belongs_to :game
+  belongs_to :player
+
```
Comment (lasley) [inline app/models/piece.rb]: Maybe a validation for color black/white?

File: spec/models/piece_spec.rb [hunk 4]
```diff
@@ -0,0 +1,5 @@
+require 'rails_helper'
+
+RSpec.describe Piece, type: :model do
+  pending "add some examples to (or delete) #{__FILE__}"
```
Comment (lasley) [inline spec/models/piece_spec.rb]: Add test for failed type validation: `type: no_exist` or similar

**Comment by ZachBerman** [comment]
Ok - so I've fixed a few things but my STI seems to be broken. I can't seem to find a resource that says to do anything differently - any advice on what I am missing?

**Pull request closed without merging**